plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -331,10 +331,12 @@ protected ResultSet getColumns(JdbcTableHandle tableHandle, DatabaseMetaData met
             throws SQLException
     {
         RemoteTableName remoteTableName = tableHandle.getRequiredNamedRelation().getRemoteTableName();
+        final String escapePattern = metadata.getSearchStringEscape();
+        final Function<String, String> escape = name -> escapeNamePattern(name, escapePattern);
         return metadata.getColumns(
-                remoteTableName.getCatalogName().orElse(null),
-                escapeNamePattern(remoteTableName.getSchemaName(), metadata.getSearchStringEscape()).orElse(null),
-                escapeNamePattern(Optional.of(remoteTableName.getTableName()), metadata.getSearchStringEscape()).orElse(null),
+                remoteTableName.getCatalogName().map(escape).orElse(null),
```

Comment:
Previously this line was

```
remoteTableName.getCatalogName().orElse(null)
```

why did you opt to apply the `escape` mapping function on the catalog name?

Reply:
IIRC JDBC specification says that `getColumns` is not accepting a pattern for catalog, so this it was not escaped in the first place.

@sergey-melnychuk Please try not to mix logical changes with refactoring (method renames, inlines, extractions etc) changes. It is easy to miss such changes. Would you like to mention this rule in https://github.com/trinodb/trino/blob/master/CONTRIBUTING.md? That way others will learn that too.

Reply:
my bad, cut-n-paste escalated quickly, preparing a fixed version now

Reply:
@kokosing , thank you for pointing out, this was my mechanical mistake, not an intention to affect the logic. Not mixing in logical changes into refactoring seems too obvious to mention :)